packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -0,0 +1,76 @@
+import type { LoadedStrapi, Entity } from '@strapi/types';
+import { omit } from 'lodash/fp';
+import { HISTORY_VERSION_UID } from '../constants';
+
+import type { HistoryVersions } from '../../../../shared/contracts';
+
+const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
+  /**
+   * Use the query engine API, not the document service,
+   * since we'll refactor history version to be just a model instead of a content type.
+   * TODO: remove this comment once the refactor is done.
+   */
+  const query = strapi.db.query(HISTORY_VERSION_UID);
+
+  let isInitialized = false;
+
+  return {
+    async init() {
+      // Prevent initializing the service twice
+      if (isInitialized) {
+        return;
+      }
+
+      strapi.documents?.middlewares.add('_all', '_all', (context, next) => {
```

Comment:
We don't plan to expose this api at the moment, but it's true that its the only way to register a middleware for every content type.

Can you add a TODO to refactor this once the `use` funciton allows registering for multiple methods at the same time?